src/services/NodeWatchService.ts

```diff
@@ -0,0 +1,74 @@
+import { networkConfig } from '@/config';
+import { NetworkType } from 'symbol-sdk';
+
+export interface NodeApiNodeInfo {
+    endpoint: string;
+    friendlyName: string;
+    mainPublicKey: string;
+    nodePublicKey: string;
+    isSslEnabled: boolean;
+    isHealthy: boolean;
+    restVersion: string;
+    wsUrl: string;
+}
+
+export class NodeWatchService {
+    private readonly url: string;
+
+    constructor(networkType: NetworkType) {
+        this.url = networkConfig[networkType].nodeWatchServiceUrl;
+    }
+
+    private mapResponseToNodeApiInfo(response: any): NodeApiNodeInfo {
+        const endpoint = response.endpoint.replace('http', 'https').replace('3000', '3001');
```

Comment:
This is likely causing problems when starting harvesting on a non-SSL node.
Steps to reproduce:
1. Start harvesting on an on-SSL node like t.rerena.nemnesia.com
2. Wallet is unable to fetch harvesting status.

The problem is that the wallet searches node by nodePublicKey/mainPublicKey and always uses the endpoint https://<node_domain>:3001, regardless of whether the node supports SSL or not. In my opinion if node does not support SSL we should use http + port 3000.

Reply:
Fixed.